workers/unity/Assets/Playground/Editor/Prefabs/PrefabPreprocessor.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using UnityEditor;
+using UnityEditor.Build;
+using UnityEditor.Build.Reporting;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace Playground.Editor
+{
+    [InitializeOnLoad]
+    public class PrefabPreprocessor : IPreprocessBuildWithReport
+    {
+        // Needed for IPreprocessBuildWithReport
+        public int callbackOrder => 0;
+
+        static PrefabPreprocessor()
+        {
+            EditorApplication.playModeStateChanged += PlaymodeStateChanged;
+        }
+
+        void IPreprocessBuildWithReport.OnPreprocessBuild(BuildReport report)
+        {
+            PreprocessPrefabs();
+        }
+
+        private static void PlaymodeStateChanged(PlayModeStateChange playModeStateChange)
```

Comment:
Nit: Can we name this `OnPlaymodeStateChanged`?

Reply:
Done